src/signalrclient/websocket_transport.cpp

```diff
@@ -74,6 +74,31 @@ namespace signalr
         // been started in which case we just stop the loop by not scheduling another receive task.
         websocket_client->receive([weak_transport, logger, receive_loop_task, weak_websocket_client](std::string message, std::exception_ptr exception)
             {
+                auto transport = weak_transport.lock();
+                if (transport)
+                {
+                    bool disconnected;
+                    {
+                        std::lock_guard<std::mutex> lock(transport->m_start_stop_lock);
+                        disconnected = transport->m_disconnected;
+                    }
+                    if (disconnected)
+                    {
+                        // stop has been called, tell it the receive loop is done and return
+                        receive_loop_task->cancel();
+                        return;
+                    }
+                }
+                else
+                {
+                    // this doesn't happen in practice, only hit by websocket tests that stop via the dtor
+                    receive_loop_task->cancel();
+                    return;
+                }
+
+                // stop waits for the receive loop to complete so the transport should never be null
+                assert(transport != nullptr);
```

Comment:
This assert seems sill now given the above code. Can we just update the tests to explicitly stop the transport, keep the assert, and drop the else?